src/session/components/avatar.rs

```diff
@@ -127,17 +129,25 @@ impl Avatar {
         let imp = self.imp();
 
         // Chat title expression
+        let type_expression = item_expression.chain_property::<Chat>("type");
         let title_expression = item_expression.chain_property::<Chat>("title");
         gtk::ClosureExpression::new::<String, _, _>(
-            &[item_expression.clone(), title_expression],
-            closure!(|_: Self, item: Option<glib::Object>, title: String| {
-                item.as_ref()
-                    .and_then(|i| i.downcast_ref::<Chat>())
-                    .filter(|chat| chat.is_own_chat())
-                    // Workaround for having a blue AdwAvatar for Saved Messages chat
-                    .map(|_| "-".to_string())
-                    .unwrap_or(title)
-            }),
+            &[item_expression.clone(), title_expression, type_expression],
```

Comment:
Well, this is actually a bit more complicated. What you actually need to pass here is the type of the user of that chat (if that chat is a Private chat). Because, otherwise, if an user will suddenly become deleted, the expression will not pick that change and the avatar's text will not change.

So, we need an `user_type_expression`. To obtain that, we'll need a `ClosureExpression` that picks the `type_expression` and that returns `Some(user)` in the case the chat is `Private` and `None` otherwise. Then you can pass that expression here.

Reply:
I hope it's solved now